src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,142 @@
+                """
+            )
+        } ?: throw LinkBuilderMethodFingerprint.exception
+
+        LinkResourceGetterFingerprint.result?.apply {
+            val instructions = this.mutableMethod.getInstructions()
+
+            // Result register of the original method call
+            var resultRegister = 0
+
+            // Remove instructions at the end and get the register of the last instruction
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Landroid/content/res/Resources;") continue
+
+                    // Get the result register of the original method call
+                    resultRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    // Remove the original method call for getting the link from resources and the move-result-object instruction
+                    this.mutableMethod.removeInstructions(index, 2)
+
+                    // Remove the instruction that uses the resultRegister as an array reference to prevent an error
+                    this.mutableMethod.removeInstructions(index - 2, 1)
+
+                    break
+                }
+            }
+
+            // Save user nickname to v7
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (
+                        methodRef.returnType != "Ljava/lang/String;") continue
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-object v7, v$sourceRegister
+                        """
+                    )
+                    break
+                }
+            }
+
+            // Call the patched method and save the result to resultRegister
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Ljava/lang/Long;") continue
+
+                    // Get the first part of the tweet ID
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+                    // Get the second part of the tweet ID
+                    val sourcePlusOne = sourceRegister + 1
```

Comment:
PLease clarify what "part" is in the comment.

Reply:
Part of the register since it's long